src/main/java/org/trigon/hardware/phoenix6/Phoenix6Inputs.java

```diff
@@ -30,15 +35,31 @@ public void toLog(LogTable table) {
         latestTable = table;
     }
 
+    /**
+     * Registers a threaded signal.
+     * Threaded signals use threading to process certain signals separately, keeping them running smoothly in the background.
+     * This avoids slowing down the main program, unlike simpler signals that run directly in it.
+     * This is also used for signals that need to be updated at a high frequency like for odometry.
+     *
+     * @param statusSignal         the threaded signal to register
+     * @param updateFrequencyHertz the frequency at which the threaded signal will be updated
+     */
     public void registerThreadedSignal(BaseStatusSignal statusSignal, double updateFrequencyHertz) {
         if (statusSignal == null || RobotHardwareStats.isReplay())
             return;
 
         registerSignal(statusSignal, updateFrequencyHertz);
-        statusSignal.setUpdateFrequency(50);
+        if (RobotHardwareStats.isSimulation()) // needed for simulation to work
+            statusSignal.setUpdateFrequency(50);
```

Comment:
explain further. 
the main problem with this is that in sim you can't run signals at high freq, and that a fast thread slows down the sim.
start with a capital letter and finish with a period